util.sh

```diff
@@ -35,6 +49,42 @@ disk_format_from_flavor () {
   esac
 }
 
+# Convert platform name to compressed artifact extension (e.g., "applehv" -> "applehv.raw.zst")
+platform_to_artifact_ext () {
+  local platform=$1
+  if [[ -z $platform ]]; then
+    echo "no platform passed"
+    return 1
+  fi
+
+  local format=$(disk_format_from_flavor "$platform")
+  echo "${platform}.${format}.zst"
+}
+
+# Get artifact extensions for a given architecture (space-separated)
+# Also includes the uncompressed "tar" artifact
+get_artifact_extensions_for_arch () {
+  local arch=$1
+  if [[ -z $arch ]]; then
+    echo "no architecture passed" >&2
+    return 1
+  fi
+
+  local platforms="${ARCH_TO_PLATFORMS[$arch]}"
+  local result=""
+
+  for platform in $platforms; do
+    result="$result $(platform_to_artifact_ext "$platform")"
+  done
+
+  # WSL also produces an uncompressed tar artifact
+  if [[ " $platforms " =~ " wsl " ]]; then
+    result="$result tar"
+  fi
```

Comment:
The comment is not right, the tar artifact is actually the coreos image as oci-archive not a disk image for a hypervisor.

As such this if condition doesn't add anything.

But also the value we add is wrong and will break the push script, the filename is `podman-machine.$arch.tar` so you are missing to add the arch part to the tar like you do for the actual disk images.

so this needs to be ` result="$result $arch.tar"`